cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/DataMigrator.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.metrics.process;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.data2.dataset2.lib.table.MetricsTable;
+import co.cask.cdap.data2.util.hbase.HBaseTableUtil;
+import co.cask.cdap.metrics.store.MetricDatasetFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Migrates data from v2 table to v3 table with salting.
+ */
+public class DataMigrator {
+  private static final Logger LOG = LoggerFactory.getLogger(DataMigrator.class);
+
+  private final LinkedHashMap<Integer, Integer> maxRecordsToScanResolutionMap;
+  private final ScheduledExecutorService scheduledExecutorService;
+  private final MigrationTableUtility migrationTableUtility;
+  private final CConfiguration cConf;
+
+
+
+  /**
+   * Data migrator to schedule data migration for metrics tables from v2 to v3 metrics tables.
+   * @param metricDatasetFactory metric dataset factory
+   * @param cConf cConfiguration
+   * @param hConf hConfiguratioin
```

Comment:
type. You could at least document this as HBase configuration.  Or remove the @param annotations for parameters that are self explanatory. Repeating the parameter name as its description is pointless.